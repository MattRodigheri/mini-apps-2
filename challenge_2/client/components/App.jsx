import React, {Component} from 'react';
import axios from 'axios';
import ChartComponent from './ChartComponent.jsx';

class App extends Component {
  constructor() {
    super();

    this.state = {
      bpiDataKeys: [],
      bpiDataValues: []
    }
  }

  componentDidMount() {
    axios.get('/coinData')
    .then((response) => {
      this.setState({
        bpiDataKeys: Object.keys(response.data.bpi),
        bpiDataValues: Object.values(response.data.bpi)
      })
    })
    .catch((error) => {
      console.log(error);
    });
  }

  render() {
    if (this.state.bpiDataKeys.length === 0) {
      return null;
    }
    return (
      <div>
        <ChartComponent data={this.state} />
        <div>Powered by CoinDesk</div>
      </div>
    )
  }
}

export default App;
