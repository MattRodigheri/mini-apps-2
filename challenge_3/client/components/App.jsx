import React, {Component} from 'react';
import axios from 'axios';

class App extends Component {
  constructor() {
    super();

    this.state = {

    }
  }

  render() {
    return (
      <div>
        Hi
      </div>
    )
  }
}

export default App;
