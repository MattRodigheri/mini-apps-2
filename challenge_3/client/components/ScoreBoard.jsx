import React, {Component} from 'react';

const ScoreBoard = function(props) {
  if (props.score.frame === 2) {
    return (
      <div>
        <div>Bowl: {props.score.firstBowl}</div>
        <div>Frame: {props.score.frame}</div>
        <div>Total Score: {props.score.score}</div>
      </div>
    )
  } else {
    return (
      <div>
        <div>Bowl: {props.score.secondBowl}</div>
        <div>Frame: {props.score.frame}</div>
        <div>Total Score: {props.score.score}</div>
      </div>
    )
  }
}

export default ScoreBoard;
