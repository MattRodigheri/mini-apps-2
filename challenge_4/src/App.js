import React, { Component } from 'react';
import './App.css';

class App extends Component {
  constructor() {
   super();

   this.state = {
     board: [
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0,
       0,0,0,0,0,0,0,0,0,0
     ]
   }
   this.reaveal = this.reveal.bind(this);
 }

 reveal(event) {
   console.log(event.target.id)
 }

  render() {
    var randomNums = [];
    while(randomNums.length < 10){
      randomNums.push(Math.floor(Math.random()*100) + 1);
    }
    return this.state.board.map((button, index) => {
      for (var i = 0; i < randomNums.length; i++) {
        if (index === randomNums[i]) {
          return <button key={index} id={'bomb'} onClick={(event) => this.reveal(event)}>b</button>
        }
      }
      return <button key={index} id={index} onClick={(event) => this.reveal(event)}></button>
    })
  }
}

export default App;
