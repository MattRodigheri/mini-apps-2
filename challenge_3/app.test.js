import React from 'React';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import App from './client/components/App.jsx';
import ScoreBoard from './client/components/ScoreBoard.jsx';


configure({adapter: new Adapter()});

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
