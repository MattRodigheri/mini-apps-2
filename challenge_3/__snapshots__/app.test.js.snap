// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
          value="0"
        >
          0
        </button>,
        <button
          onClick={[Function]}
          value="1"
        >
          1
        </button>,
        <button
          onClick={[Function]}
          value="2"
        >
          2
        </button>,
        <button
          onClick={[Function]}
          value="3"
        >
          3
        </button>,
        <button
          onClick={[Function]}
          value="4"
        >
          4
        </button>,
        <button
          onClick={[Function]}
          value="5"
        >
          5
        </button>,
        <button
          onClick={[Function]}
          value="6"
        >
          6
        </button>,
        <button
          onClick={[Function]}
          value="7"
        >
          7
        </button>,
        <button
          onClick={[Function]}
          value="8"
        >
          8
        </button>,
        <button
          onClick={[Function]}
          value="9"
        >
          9
        </button>,
        <button
          onClick={[Function]}
          value="10"
        >
          10
        </button>,
        <ScoreBoard
          score={
            Object {
              "firstBowl": 0,
              "frame": 1,
              "frameHalf": 1,
              "score": 0,
              "secondBowl": 0,
              "spareBonus": 0,
              "strikeBonus": 0,
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "0",
          "onClick": [Function],
          "value": "0",
        },
        "ref": null,
        "rendered": "0",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1",
          "onClick": [Function],
          "value": "1",
        },
        "ref": null,
        "rendered": "1",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2",
          "onClick": [Function],
          "value": "2",
        },
        "ref": null,
        "rendered": "2",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "3",
          "onClick": [Function],
          "value": "3",
        },
        "ref": null,
        "rendered": "3",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "4",
          "onClick": [Function],
          "value": "4",
        },
        "ref": null,
        "rendered": "4",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "5",
          "onClick": [Function],
          "value": "5",
        },
        "ref": null,
        "rendered": "5",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "6",
          "onClick": [Function],
          "value": "6",
        },
        "ref": null,
        "rendered": "6",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "7",
          "onClick": [Function],
          "value": "7",
        },
        "ref": null,
        "rendered": "7",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "8",
          "onClick": [Function],
          "value": "8",
        },
        "ref": null,
        "rendered": "8",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "9",
          "onClick": [Function],
          "value": "9",
        },
        "ref": null,
        "rendered": "9",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "10",
          "onClick": [Function],
          "value": "10",
        },
        "ref": null,
        "rendered": "10",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "score": Object {
            "firstBowl": 0,
            "frame": 1,
            "frameHalf": 1,
            "score": 0,
            "secondBowl": 0,
            "spareBonus": 0,
            "strikeBonus": 0,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
            value="0"
          >
            0
          </button>,
          <button
            onClick={[Function]}
            value="1"
          >
            1
          </button>,
          <button
            onClick={[Function]}
            value="2"
          >
            2
          </button>,
          <button
            onClick={[Function]}
            value="3"
          >
            3
          </button>,
          <button
            onClick={[Function]}
            value="4"
          >
            4
          </button>,
          <button
            onClick={[Function]}
            value="5"
          >
            5
          </button>,
          <button
            onClick={[Function]}
            value="6"
          >
            6
          </button>,
          <button
            onClick={[Function]}
            value="7"
          >
            7
          </button>,
          <button
            onClick={[Function]}
            value="8"
          >
            8
          </button>,
          <button
            onClick={[Function]}
            value="9"
          >
            9
          </button>,
          <button
            onClick={[Function]}
            value="10"
          >
            10
          </button>,
          <ScoreBoard
            score={
              Object {
                "firstBowl": 0,
                "frame": 1,
                "frameHalf": 1,
                "score": 0,
                "secondBowl": 0,
                "spareBonus": 0,
                "strikeBonus": 0,
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "0",
            "onClick": [Function],
            "value": "0",
          },
          "ref": null,
          "rendered": "0",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1",
            "onClick": [Function],
            "value": "1",
          },
          "ref": null,
          "rendered": "1",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2",
            "onClick": [Function],
            "value": "2",
          },
          "ref": null,
          "rendered": "2",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3",
            "onClick": [Function],
            "value": "3",
          },
          "ref": null,
          "rendered": "3",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "4",
            "onClick": [Function],
            "value": "4",
          },
          "ref": null,
          "rendered": "4",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "5",
            "onClick": [Function],
            "value": "5",
          },
          "ref": null,
          "rendered": "5",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "6",
            "onClick": [Function],
            "value": "6",
          },
          "ref": null,
          "rendered": "6",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "7",
            "onClick": [Function],
            "value": "7",
          },
          "ref": null,
          "rendered": "7",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "8",
            "onClick": [Function],
            "value": "8",
          },
          "ref": null,
          "rendered": "8",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "9",
            "onClick": [Function],
            "value": "9",
          },
          "ref": null,
          "rendered": "9",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "10",
            "onClick": [Function],
            "value": "10",
          },
          "ref": null,
          "rendered": "10",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "score": Object {
              "firstBowl": 0,
              "frame": 1,
              "frameHalf": 1,
              "score": 0,
              "secondBowl": 0,
              "spareBonus": 0,
              "strikeBonus": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
